docs/rfcs/2023-03-29-file-external-table/rfc.md

```diff
@@ -0,0 +1,86 @@
+---
+Feature Name: "File external table"
+Tracking Issue: https://github.com/GreptimeTeam/greptimedb/issues/1041
+Date: 2023-03-08
+Author: "Xu Wenkang <wenymedia@gmail.com>"
+---
+
+File external table
+
+---
+
+# Summary
+
+Allows users to perform SQL queries on files
+
+# Motivation
+
+User data may already exist in other storages, i.g., file systems/s3, etc. in CSV, parquet, JSON format, etc. We can provide users the ability to perform SQL queries on these files.
+
+
+# Details
+## Overview
+![overview](external-table-engine-overview.png)
+
+We implement a file table engine that creates an external table by accepting user-specified file paths and treating all records as immutable.
+
+1. File Format Decoder: decode files to the `RecordBatch` stream.
+2. Table engine: stores `RecordBatch` in memory and provides scan ability.
+
+## Life cycle
+
+### Register table
+1. Creates metadata and stores it in memory.
+2. Creates the table via file table engine.
+3. Registers table to `CatalogProvider` and it won't register table to `SystemCatalog` (a place that will persist all user-created tables).
+4. Registers table to `SessionLayer` (currently not exists).
+
+### Clean up
+For now, GreptimeDB still lacks a session layer. The system can't release an external table automatically when the user session is closed. But it still has serval ways to allow users to release an external table. 
```

Comment:
Is it true? @MichaelScofield I meant that do we still lacks a session layer? I remember that we already implement it.

Reply:
Is external table still temporary?

Reply:
> Is external table still temporary?

I'm going to make it durable again 🤣

Reply:
> If we align to Clickhouse's concept, external table is ephermeral. Does mysql or pg has external tables as well? Are they durable?

I think the file external table in our system might be more similar to the Clickhouse [integrations engine](https://clickhouse.com/docs/en/engines/table-engines/integrations) (and StarRocks [file external table](https://docs.starrocks.io/en-us/latest/data_source/file_external_table)) and not the [external data](https://clickhouse.com/docs/en/engines/table-engines/special/external-data). Mysql has the "create table externally", It's durable but not what we wanted. PostgreSQL has foreign tables, and it's durable.